### PR TITLE
Detect user disconnects and dispose markers for websocket sample.

### PR DESCRIPTION
**Pull request by navilperez**
Websocket server will attach the workspaceId to each user in order track
users and send a broadcast to users when someone disconnects. Upon
disconnecting, the user is removed from the database and the event is
broadcast to the other users.

UserDataManager disposes of markers that have disconnected.

### CODE REVIEW

File: blockly-rtc/server/websocket/users_handlers.js [hunk 1]
```diff
@@ -53,6 +53,43 @@ async function getPositionUpdatesHandler(workspaceId, callback) {
...
+
+/**
+ * Handler for a disconnect. Delete the user from the users table.
+ * @param {string} workspaceId The workspaceId for the disconnecting user.
+ * @param {!Function} callback The callback that broadcasts the disconnect to
+ * the connected users.
+ * @public
+ */
+async function disconnectUserHandler(workspaceId, callback) {
+  console.log('User disconnected.');
```
Comment (rachel-fenichel) [inline blockly-rtc/server/websocket/users_handlers.js]: Remove log.

File: blockly-rtc/src/UserDataManager.js [hunk 2]
```diff
@@ -49,7 +50,24 @@ export default class UserDataManager {
     } else {
       this.pollServer_();
     };
-};
+    if (this.getUserDisconnects) {
+      this.getUserDisconnects(this.disposeMarker_.bind(this));
```
Comment (rachel-fenichel) [inline blockly-rtc/src/UserDataManager.js:54]: Won't this always be null here?

**Review by rachel-fenichel — COMMENTED** [review]
One question, otherwise lgtm.

File: blockly-rtc/src/UserDataManager.js (hunk 2 above)
Comment (navilperez) [reply blockly-rtc/src/UserDataManager.js:54]: The current flow is that `enablePresence()` assigns `this.getUserDisconnects`. And then `start()` is called which calls the function.

Another approach is to assign and call the function `this.getUserDisconnects` in `enablePresence` and remove L53-54 from `start`.

I am open and happy with either option.

Comment (navilperez) [reply blockly-rtc/src/UserDataManager.js:54]: Correction, when I say `enablePresence` above, I am actually referring to `setPresenceHandlers`

**Comment by rachel-fenichel** [comment]
Approve once you remove the leftover `console.log`.

**Pull request merged**